resolver/dns/dns_resolver.go

```diff
@@ -242,7 +245,19 @@ func (d *dnsResolver) watcher() {
 		case <-d.t.C:
 		case <-d.rn:
 		}
+
 		result, sc := d.lookup()
+		if !d.t.Stop() {
```

Comment:
Oh, I see what's going on here.  How about moving this `Stop` call to inside the `case <-d.rn` above instead?

Reply:
Done.